src/DependencyResolver/Package.php

```diff
@@ -129,25 +128,28 @@ public function githubOrgAndRepository(): string
     }
 
     /**
-     * @param list<string> $input
+     * @param list<string>|null $input
      *
-     * @return list<OperatingSystemFamily>
+     * @return non-empty-list<OperatingSystemFamily>|null
      */
-    private static function convertInputStringsToOperatingSystemFamilies(array $input): array
+    private static function convertInputStringsToOperatingSystemFamilies(array|null $input): array|null
     {
+        if ($input === null) {
+            return null;
+        }
+
         $osFamilies = [];
         foreach ($input as $value) {
             // try to normalize a bit the input
             $valueToTry = ucfirst(strtolower($value));
```

Comment:
One improvement I didn't do, but I think it could make sense to **always** lowercase (including the `enum` values), to avoid case sensitivity nonsense, as this approach isn't flawless.